RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractEdge.java

```diff
@@ -0,0 +1,197 @@
+import org.sleuthkit.autopsy.ingest.DataSourceIngestModuleProcessTerminator;
+import org.sleuthkit.autopsy.ingest.IngestJobContext;
+import org.sleuthkit.autopsy.ingest.IngestServices;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.Content;
+import org.sleuthkit.datamodel.TskCoreException;
+
+public class ExtractEdge extends Extract{
+    
+    private static final Logger logger = Logger.getLogger(ExtractIE.class.getName());
+    private final IngestServices services = IngestServices.getInstance();
+    private final String moduleTempResultsDir;
+    private Content dataSource;
+    private IngestJobContext context;
+    
+    private static String ESE_TOOL_NAME = "ESEDatabaseView.exe";
+    private static File ESE_TOOL_FILE;
+    private static String EDGE_WEBCACHE_NAME = "WebCacheV01.dat";
+    private static String EDGE = "Edge";
+
+    ExtractEdge() throws NoCurrentCaseException{
+        moduleName = NbBundle.getMessage(Chrome.class, "ExtractEdge.moduleName");
+        moduleTempResultsDir = RAImageIngestModule.getRATempPath(Case.getCurrentCaseThrows(), EDGE) + File.separator + "results";
+    }
+    
+    @Override
+    void process(Content dataSource, IngestJobContext context) {
+        this.dataSource = dataSource;
+        this.context = context;
+        dataFound = false;
+        
+        this.processWebCache();
+        
+         // Bookmarks come from spartan.edb different file
+        this.getBookmark(); // Not implemented yet
+    }
+    
+    void processWebCache(){
+        Path path = Paths.get("ESEDatabaseView", ESE_TOOL_NAME);
+        ESE_TOOL_FILE =  InstalledFileLocator.getDefault().locate(path.toString(), ExtractEdge.class.getPackage().getName(), false); //NON-NLS
+        if (ESE_TOOL_FILE == null) {
+            this.addErrorMessage(
+                    NbBundle.getMessage(this.getClass(), "ExtractEdge.process.errMsg.unableFindESEViewer", this.getName()));
+            logger.log(Level.SEVERE, "Error finding ESEDatabaseViewer program "); //NON-NLS
+        }
+        
+         final String esedumper = ESE_TOOL_FILE.getAbsolutePath();
+
+        // get WebCacheV01.dat files
+        org.sleuthkit.autopsy.casemodule.services.FileManager fileManager = currentCase.getServices().getFileManager();
+        List<AbstractFile> webCachFiles;
+        try {
+            webCachFiles = fileManager.findFiles(dataSource, EDGE_WEBCACHE_NAME); //NON-NLS
+        } catch (TskCoreException ex) {
+            this.addErrorMessage(NbBundle.getMessage(this.getClass(), "ExtractEdge.process.errMsg.errGettingWebCacheFiles",
+                    this.getName()));
+            logger.log(Level.WARNING, "Error fetching 'index.data' files for Internet Explorer history."); //NON-NLS
+            return;
+        }
+
+        if (webCachFiles.isEmpty()) {
+            String msg = NbBundle.getMessage(this.getClass(), "ExtractEdge.process.errMsg.noWebCachFiles");
+            logger.log(Level.INFO, msg);
```

Comment:
Log messages should never be localized. The most likely consumer of them is someone on the Basis dev team and we speak English; English is also a de facto lingua franca, so there is no up side in making logs the primarily English-speaking Basis devs cannot read.